src/main/java/com/conveyal/taui/controllers/WorkerController.java

```diff
@@ -206,14 +213,49 @@ private String jsonResponse (Response response, int statusCode, Object object) {
      * Fetch status of all unfinished jobs as a JSON list.
      */
     private String getAllJobs(Request request, Response response) {
-        return jsonResponse(response, HttpStatus.OK_200, broker.getJobSummary());
+        String accessGroup = request.attribute("accessGroup");
+        if (!AnalysisServerConfig.adminAccessGroup.equals(accessGroup)) {
+            response.status(401);
+            return "You do not have access.";
+        }
+
+        Collection<JobStatus> jobStatuses = broker.getJobSummary();
+        for (JobStatus jobStatus : jobStatuses) {
+            if (!jobStatus.jobId.equals("SUM")) {
+                jobStatus.regionalAnalysis = Persistence.regionalAnalyses
+                        .find(QueryBuilder.start("_id").is(jobStatus.jobId).get()).next();
+            }
+        }
+
+        return jsonResponse(response, HttpStatus.OK_200, jobStatuses);
     }
 
     /**
      * Report all workers that have recently contacted the broker as JSON list.
      */
     private String getAllWorkers(Request request, Response response) {
-        return jsonResponse(response, HttpStatus.OK_200, broker.getWorkerObservations());
+        String accessGroup = request.attribute("accessGroup");
+        if (!AnalysisServerConfig.adminAccessGroup.equals(accessGroup)) {
+            response.status(401);
+            return "You do not have access.";
+        }
+
+        Collection<WorkerObservation> observations = broker.getWorkerObservations();
+        Map<String, Bundle> bundleForNetworkId = new HashMap<>();
```

Comment:
It's worth a comment here to state clearly that this is just a cache to reduce database queries. Otherwise it takes the reader a while to verify that there is nothing more complex going on here - just lazy-loading a cache of Bundle values.